spiff/models.py

```diff
@@ -1,15 +1,22 @@
+import logging
 from abc import ABC, abstractmethod
 from typing import Callable, List, Type
 
 import torch
 import torch_geometric.nn.models
 import torch_geometric.nn.models.basic_gnn as basic_gnn
 
+logger = logging.getLogger(__name__)
+
+
+class ModelFactory(ABC):
+    @abstractmethod
+    def __call__(self, *args) -> torch.nn.Module:

```

Comment:
This should not take any arguments or only very general arguments (input_size, output_size). Specific arguments should be passed as arguments of the factory __init__.